### PR TITLE
Login integration

### PR DESCRIPTION
**Pull request by junron**
[Task](https://codein.withgoogle.com/dashboard/task-instances/4780640122699776/)

Added login integration based on backend by @Khukhuna, login UI by myself
Disables sign in key until form is valid
Adds validation for each of the fields instead of the whole form.

### CODE REVIEW

File: src/app/services/authentication.service.ts [hunk 1]
```diff
@@ -1,17 +1,32 @@
...
 
-  private getUserInfo(){
+  public loginUser(user: User){
     //authenticates user while login attempt
+    const options = {
+      headers: new HttpHeaders({
+        "Content-Type":"application/json",
+      })
+    }
+    return this.httpClient.post(`${this.apiUrl}/api-token-auth/`,JSON.stringify(user),options)
```
Comment (Khukhuna) [inline src/app/services/authentication.service.ts]: I might be wrong, but I think our backend does not accept json format. Did you test this?

Comment (junron) [reply src/app/services/authentication.service.ts]: I had a problem creating users with your backend, the picture in your pull request does not indicate the user's password, so I was unable to create users and test it. Was just in the process of asking this on your pr

Comment (junron) [reply src/app/services/authentication.service.ts]: Could you help me on this?

Comment (jatinkatyal13) [reply src/app/services/authentication.service.ts]: @Khukhuna backend does accept application/json!

**Comment by Khukhuna** [comment]
Try:
**Username:** test
**Password:**  jbossoutreach

You can send data as:
application/x-www-form-urlencoded

File: src/app/services/authentication.service.ts (hunk 1 above)
Comment (Khukhuna) [reply src/app/services/authentication.service.ts]: Sorry for late reply, I'm in school atm, Try:
**Username:** test
**Password:**  jbossoutreach

You can send data as:
application/x-www-form-urlencoded

**Comment by junron** [comment]
I will try that soon

File: src/app/services/authentication.service.ts (hunk 1 above)
Comment (junron) [reply src/app/services/authentication.service.ts]: That doesn't seem to work, i get 
```js
{
    "non_field_errors": [
        "Unable to log in with provided credentials."
    ]
}
```

when using either `application/json` or `application/x-www-form-urlencoded`.

Comment (junron) [reply src/app/services/authentication.service.ts]: I tried
```js
{
	"user":{
		"username":"abcde",
		"first_name":"jun",
		"last_name":"ron",
		"email":"fake@fake.com",
		"password":"abcde"
	},
	"name":"abcde",
	"password":"abcde"
}
```
for creating users, it seems to have no error, but login still doesn't work

Comment (jatinkatyal13) [reply src/app/services/authentication.service.ts]: Actually creating users is still in progress! There are a few tweaks that are left but I have given login details to @nikitadhiman for student and organization! Until we fix the registration system you can use those credentials

**Comment by nikitadhiman** [comment]
Should I put the details in comment here?

**Comment by Khukhuna** [comment]
You are sending request to my heroku deploy not jboss own.

My branch is not up to date with JBoss repo, so sign does not work on my heroku deploy. Use jboss instead

**Comment by jatinkatyal13** [comment]
> Should I put the details in comment here?

I don't know if that would be a great idea! I guess just share with participants in the gitter channel!

**Comment by nikitadhiman** [comment]
Yup, I was thinking the same.guys you can ping me on gitter for details. Thanks

**Comment by junron** [comment]
I get the following error after using the new url:
`Access to XMLHttpRequest at '<the_domain>api-token-auth/' from origin 'http://localhost:4200' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

I have set the `Access-Control-Allow-Origin` as `*`

**Comment by junron** [comment]
Squashed commits

**Pull request closed without merging**